src/message/connection.js

```diff
@@ -531,7 +531,8 @@ Connection.prototype._tryReconnect = function () {
   } else {
     this._clearReconnect()
     this.close()
-    this._client.emit(C.MAX_RECONNECTION_ATTEMPTS_REACHED, this._reconnectionAttempt)
+    this._setState(C.CONNECTION_STATE.CLOSED)
```

Comment:
Hi @jdmnd 
I've accepted the license.
regarding the state change we're using JWT token authentication. We always need to ensure fresh tokens (lifetime 1min or 5min) during auth.

First we were not using the embedded reconnect mechanism: _options.maxReconnectAttempts = 1. Reconnection was handled from outside the ds-io.client, so we needed the connection to end in state CLOSED instead of ERROR. Connection CLOSED allowed us to call login() on the same deepstream instance with a fresh token.  That's why we added the _setState(C.CONNECTION_STATE.CLOSED) when MAX_RECONNECTION_ATTEMPTS_REACHED.

Now we've changed our implementation and started using the embedded reconnection mechanism (retry forever). Right after connection changes to AWAITING_AUTHENTICATION, we set _connection._authParams to null. This gives us the time we need to go get a fresh token and then call login.

We're not using the _setState anymore. If you think it's better to leave the connection in the ERROR state pls let me know and i'll update the pr. 

Thanks and sorry for the lengthy comment.